scriptmodules/ports/hexen2.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="hexen2"
+rp_module_desc="Hexen II - Hammer of Thyrion source port"
+rp_module_licence="GLP2"
+rp_module_help="For registered version, please add your full version PAK files to $romdir/ports/hexen2/data1/ to play. These files for the registered version are required: pak0.pak, pak1.pak and strings.txt. The registered pak files must be patched to 1.11 for Hammer of Thyrion."
+rp_module_section="opt"
+rp_module_flags=""
+
+
+# Get depends required by hexen2
+function depends_hexen2() {
+    getDepends cmake libsdl1.2debian libsdl-image1.2 libsdl-mixer1.2 libsdl-mixer1.2-dev libsdl-net1.2 libsdl-net1.2-dev timidity
+}
+
+# Clone from uhexen2 source
+function sources_hexen2() {
+    gitPullOrClone "$md_build" https://github.com/svn2github/uhexen2.git
+}
+
+# Build hexen2
+function build_hexen2() {
+    cd "$md_build/engine/hexen2"
+    ./build_all.sh
+    md_ret_require="$md_build/engine/hexen2/hexen2"
+}
+
+function install_hexen2() {
+    md_ret_files=(        
+        'engine/hexen2/hexen2'
+    )
+}
+
+# Download hexen2 shareware version
+function game_data_hexen2() {
+	if [[ ! -f "$romdir/ports/hexen2/pak0.pak" ]]; then
+        downloadAndExtract "https://netix.dl.sourceforge.net/project/uhexen2/Hexen2Demo-Nov.1997/hexen2demo_nov1997-linux-i586.tgz" "$romdir/ports/hexen2"
+		mv /home/pi/RetroPie/roms/ports/hexen2/hexen2demo_nov1997/data1 /home/pi/RetroPie/roms/ports/hexen2
+		rm -r /home/pi/RetroPie/roms/ports/hexen2/hexen2demo_nov1997
+    fi
+    # always chown as moveConfigDir in the configure_ script would move the root owned demo files
+    chown -R $user:$user "$romdir/ports/hexen2"
+}
+
+# Configure hexen2 RetroPie
+function configure_hexen2() {
+    mkRomDir "ports"
+    mkRomDir "ports/hexen2"
+
+    mkUserDir "$home/.hexen2"
+    
+        moveConfigDir "$home/.hexen2" "$romdir/ports/hexen2"
+
+    # Add user for no sudo run
+    usermod -a -G video $user
+
+    [[ "$md_mode" == "install" ]] && game_data_hexen2
+        
+        addPort "$md_id" "hexen2" "Hexen II" "$md_inst/hexen2"
```

Comment:
also here, wrong indentation

Reply:
also, how the data files are passed to the executable file? is that not parammetrisable?

Reply:
The source port is being compiled with SDL from the build_all_sh located in ~/engine/hexen2 and links to an hexen2 binary file once it is complete. The source port does not create a config folder when compiling. 

It only creates the .hexen2 config folder when it can not locate the gamedata (pak files) that the source port need to play the game. notice this happens when you try to run hexen2 binary file.

That is why i have included the "mkUserDir "$home/.hexen2" "$romdir/ports/hexen2" at line 59. This is for making it easier to replace the demo pak files with the registered version as with Quake 3 RetroPie folder in $romdir/ports/quake3.

Reply:
It means that that Hammer of Thyrion source port will need the demo pak files or registered pak files in its config folder and the pak files must be placed in this dir: ~/.hexen2/data1/ for the hexen2 binary to load them.

Reply:
> That is why i have included the "mkUserDir "$home/.hexen2" "$romdir/ports/hexen2" at line 59. This is for making it easier to replace the demo pak files with the registered version as with Quake 3 RetroPie folder in $romdir/ports/quake3.

Sorry, I'm still confused. The gamedata (pak files) should always be in `$romdir/ports/hexen2` and never be searched or stored in `$HOME/.hexen2`. Is this a limitation of the game binary?

> It means that that Hammer of Thyrion source port will need the demo pak files or registered pak files in its config folder and the pak files must be placed in this dir: ~/.hexen2/data1/ for the hexen2 binary to load them.

Do you mean that the game files must go to `$HOME/.hexen2/data1`?  What files would go then in `$romdir/ports/hexen2`?

Maybe easier to understand if you detail step-by-step how you would install and popualte the game files manually, i.e. what files goes where.

Reply:
Yes that is exactly what i mean.

1. When you download/clone from git the install dir will possibly look like this: uhexen2/ or as $md_build/
2. You change dir to $md_build/engine/hexen2 and run the build_all_sh to build the game with its SDL libs.
3. After building the source port from the build_all_sh, it will have linked to an hexen2 binary file.
4. When you for the first time try to launch the hexen2 binary file it will tell you that it can not locate the game files and will create a config folder called in this dir /$home/.hexen2/data1 and you must place the game files (eg. pak0.pak, hexen.rc and such) into this folder for the binary file to load the game files.
5. The reason for creating "$romdir/ports/hexen2" is to symlink it to the config folder: /$home/.hexen2/data1 so that it will be easier to replace the demo files to the registered pak files. **Notice this is the same procedure that Quake3 for RetroPie uses. I borrowed this from the Quake3 scriptmodule.**

Finally it is important to mention that the Hammer of Thyrion source port require either patched 1.11 registered pak files(pak0.pak, pak1.pak) or the Hexen2Nov1997 demo pak files to be played.

I hope this explanation is sufficient enough. Otherwise feel free to ask. :-)